### PR TITLE
Add a juju action to create a docker registry inside the cluster

### PR DESCRIPTION
**Pull request by Cynerva**
**What this PR does / why we need it**:

This PR adds a "registry" action to the kubernetes-worker juju charm. It allows to easily deploy a private, authenticated, TLS-protected docker registry in your kubernetes cluster.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

Original PR at https://github.com/juju-solutions/kubernetes/pull/108. I have rebased this on behalf of @jacekn.

**Release note**:

```release-note
The kubernetes-worker Juju charm now has a "registry" action that greatly simplifies the deployment of a private, authenticated, TLS-protected docker registry in your kubernetes cluster. See the README for usage examples
```

### CODE REVIEW

File: cluster/juju/layers/kubernetes-worker/templates/registry.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+apiVersion: v1
+kind: Secret
+metadata:
+  name: registry-auth-data
+type: Opaque
+data:
+  htpasswd: {{ htpasswd }}
+---
+apiVersion: v1
+kind: ReplicationController
```
Comment (caio1982) [inline cluster/juju/layers/kubernetes-worker/templates/registry.yaml]: IMHO it's worth making it a full Deployment object.

**Comment by mbruzek** [comment]
@k8s-bot ok to test

**Comment by mbruzek** [comment]
The k8s bot does not test juju so I doubt the failure was related to this change. Going to kick the tests again.

@k8s-bot gce etcd3 e2e test this

**Comment by caio1982** [comment]
Thanks folks, that address the Docker registry problems we had been seeing, it does look good to me.

**Comment by Cynerva** [comment]
Closing, superceded by https://github.com/kubernetes/kubernetes/pull/44500

**Pull request closed without merging**